assets/tests/components/lateView.test.tsx

```diff
@@ -175,4 +182,602 @@ describe("LateView", () => {
+
+    act(() => {
+      wrapper
+        .find(
+          '.m-late-view__late-buses .m-late-view__data-row--unselected input[type="checkbox"]'
+        )
+        .last()
+        .simulate("click")
+    })
+    expect(!isIndeterminate(wrapper, lateMasterCheckboxSelector))
+    expect(isChecked(wrapper, lateMasterCheckboxSelector))
+
+    expect(!isIndeterminate(wrapper, lateMasterCheckboxSelector))
+    expect(!isChecked(wrapper, lateMasterCheckboxSelector))
+
+    act(() => {
+      wrapper
+        .find(
+          '.m-late-view__missing-logons .m-late-view__data-row--unselected input[type="checkbox"]'
+        )
+        .first()
+        .simulate("click")
+    })
+    expect(isIndeterminate(wrapper, missingMasterCheckboxSelector))
+    expect(!isChecked(wrapper, missingMasterCheckboxSelector))
+
+    act(() => {
+      wrapper
+        .find(
+          '.m-late-view__missing-logons .m-late-view__data-row--unselected input[type="checkbox"]'
+        )
+        .last()
+        .simulate("click")
+    })
+    expect(!isIndeterminate(wrapper, missingMasterCheckboxSelector))
+    expect(isChecked(wrapper, missingMasterCheckboxSelector))
+  })
+
+  test("master checkbox toggles multiple rows", () => {
+    const lateVehicle1 = vehicleFactory.build({
+      routeId: "route",
+      runId: "run1",
+      scheduleAdherenceSecs: 901,
+    })
+    const lateVehicle2 = vehicleFactory.build({
+      routeId: "route",
+      runId: "run2",
+      scheduleAdherenceSecs: 901,
+    })
+    const missingLogonGhost1 = ghostFactory.build({
+      routeId: "route",
+      runId: "run3",
+      scheduledLogonTime: 15301,
+    })
+    const missingLogonGhost2 = ghostFactory.build({
+      routeId: "route",
+      runId: "run4",
+      scheduledLogonTime: 15301,
+    })
+
+    const vehiclesByRouteId = {
+      route: [
+        lateVehicle1,
+        lateVehicle2,
+        missingLogonGhost1,
+        missingLogonGhost2,
+      ],
+    }
+
+    const wrapper = mount(
+      <StateDispatchProvider state={state} dispatch={jest.fn()}>
+        <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
+          <LateView />
+        </VehiclesByRouteIdProvider>
+      </StateDispatchProvider>
+    )
+
+    expect(
+      wrapper.find(
+        '.m-late-view__late-buses .m-late-view__data-row--selected input[type="checkbox"]'
+      )
+    ).toHaveLength(0)
+
+    act(() => {
+      wrapper.find(lateMasterCheckboxSelector).first().simulate("click")
+    })
+    expect(
+      wrapper.find(
+        '.m-late-view__late-buses .m-late-view__data-row--selected input[type="checkbox"]'
+      )
+    ).toHaveLength(2)
+
+    act(() => {
+      wrapper
+        .find('.m-late-view__data-row--selected input[type="checkbox"]')
+        .first()
+        .simulate("click")
+    })
+    expect(
+      wrapper.find(
+        '.m-late-view__late-buses .m-late-view__data-row--selected input[type="checkbox"]'
+      )
+    ).toHaveLength(1)
```

Comment:
**suggestion:** I think it would also make sense to make an assertion about the state of the master checkbox after unchecking the individual late row.